### PR TITLE
[WIP - Tests Pending] ROC-3258: Judgement Request payment by instalment

### PR DESCRIPTION
**Pull request by varma02**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3258 

### Change description ###

Judgement Request payment by instalment changes in pdf

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html [hunk 1]
```diff
@@ -239,12 +243,33 @@ <h2>Request for judgment by default</h2>
...
       </tr>
-      <tr>
-        <td>When you want the defendant to pay:</td>
-        <td>
-          {{ ccj.repaymentOption }}
-        </td>
-      </tr>
+      {% if ccj.repaymentOption is not empty %}
+        <tr>
+          <td>How do you want the defendant to pay:</td>
```
Comment (geek-soft) [inline src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html]: Is there a `do` in the prototype?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/RepaymentPlanContentProvider.java [hunk 2]
```diff
@@ -1,45 +1,17 @@
...
-    private static String getRepaymentPlanContent(RepaymentPlan repaymentPlan) {
-        return String.format(
-            "first payment of %s on %s. This will be followed by %s %s",
-            formatMoney(repaymentPlan.getFirstPayment()),
-            formatDate(repaymentPlan.getFirstPaymentDate()),
-            formatMoney(repaymentPlan.getInstalmentAmount()),
-            repaymentPlan.getPaymentSchedule().getDescription()
-        );
+    public static Optional<RepaymentPlan> create(CountyCourtJudgment countyCourtJudgment) {
+        return countyCourtJudgment.getRepaymentPlan();
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/RepaymentPlanContentProvider.java]: That is wrong. You basically passing raw data into PDF generator, if we wanted to do it (which we do not) we would remove `ContentProvider` completely.

To give you some background ContentProvider is an objects that reformats data for displaying in PDF's due to the fact PDF generator does not have any formatting facilitates. To give you an example we need to format date to UK format in PDF so passing raw `2018-01-01` would not be enough, in that case ContentProvider formats it to be `1 January 2018`, the form ready for being displayed in PDF.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/CCJContent.java [hunk 3]
```diff
@@ -26,9 +31,12 @@ public CCJContent(Claim claim, InterestCalculationService interestCalculationSer
         this.amount = new AmountContentProvider(interestCalculationService).create(claim);
         claim.getCountyCourtJudgment().getDefendantDateOfBirth()
             .ifPresent((dateOfBirth -> this.defendantDateOfBirth = formatDate(dateOfBirth)));
-        this.repaymentOption = RepaymentPlanContentProvider.create(claim.getCountyCourtJudgment());
+        this.repaymentOption = RepaymentPlanContentProvider.create(claim.getCountyCourtJudgment()).orElse(null);
         this.requestedAt = Formatting.formatDateTime(claim.getCountyCourtJudgmentRequestedAt());
         this.requestedDate = formatDate(claim.getCountyCourtJudgmentRequestedAt());
+        this.paymentOption = claim.getCountyCourtJudgment().getPaymentOption();
+        this.firstInstalmentPaymentDate = formatDate(repaymentOption.getFirstPaymentDate());
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/CCJContent.java]: Please move it into `RepaymentPlanContentProvider`.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/CCJContent.java [hunk 4]
```diff
@@ -26,9 +31,12 @@ public CCJContent(Claim claim, InterestCalculationService interestCalculationSer
...
+        this.paymentFrequency = repaymentOption.getPaymentSchedule().getDescription();
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/models/CCJContent.java]: Please move it into `RepaymentPlanContentProvider`.

File: src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html [hunk 5]
```diff
@@ -239,12 +243,33 @@ <h2>Request for judgment by default</h2>
...
+          <td>
+            {{ ccj.paymentOption }}
```
Comment (geek-soft) [inline src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html]: That option is independent from repayment plan, don't we want do display it for all payment options like:

```
IMMEDIATELY,
FULL_BY_SPECIFIED_DATE,
INSTALMENTS;
```

**Comment by geek-soft** [comment]
@varma02 can I ask you please to enclose example PDF generated by the app to the PR so I can have a look without ruining the app?

File: src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html (hunk 1 above)
Comment (varma02) [reply src/main/resources/staff/templates/document/countyCourtJudgmentDetails.html]: yes, there is do in prototype

**Comment by varma02** [comment]
[Defendant-000MC149-county-court-judgment-details.pdf](https://github.com/hmcts/cmc-claim-store/files/1821081/Defendant-000MC149-county-court-judgment-details.pdf)


Attached is pdf after changes

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/RepaymentPlanContentProviderTest.java [hunk 6]
```diff
@@ -19,6 +19,7 @@ public void createImmediatePaymentOption() {
             .withPaymentOption(PaymentOption.IMMEDIATELY)
             .build();
 
+
```
Comment (timja-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/RepaymentPlanContentProviderTest.java]: remove this line please, too much space

Comment (varma02) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/RepaymentPlanContentProviderTest.java]: ok :), but i haven't finished working on tests yet "[WIP - Tests Pending] ROC-3258: Judgement Request payment by instalment"

**Comment by varma02** [comment]
i will create new pull request with all corrections

**Pull request closed without merging**